forest/src/cli/chain_cmd.rs

```diff
@@ -0,0 +1,126 @@
+    /// blockstore and returns raw bytes
+    #[structopt(about = "<CID> Read the raw bytes of an object")]
+    ReadObj {
+        #[structopt(help = "Input a valid CID")]
+        cid: String,
+    },
+}
+
+impl ChainCommands {
+    pub async fn run(&self) {
+        // TODO handle cli config
+        match self {
+            Self::Block { cid } => {
+                let cid: Cid = cid.parse().unwrap();
+                let client = new_client();
+
+                let blk = block(client, cid)
+                    .await
+                    .map_err(|e| {
+                        stringify_rpc_err(e);
+                    })
+                    .unwrap();
+                println!("{}", serde_json::to_string_pretty(&blk).unwrap());
+            }
+            Self::Genesis => {
+                let client = new_client();
+
+                let gen = genesis(client)
+                    .await
+                    .map_err(|e| {
+                        stringify_rpc_err(e);
+                    })
+                    .unwrap();
+                println!("{}", serde_json::to_string_pretty(&gen).unwrap());
+            }
+            Self::Head => {
+                let client = new_client();
+
+                let canonical = head(client)
+                    .await
+                    .map_err(|e| {
+                        stringify_rpc_err(e);
+                    })
+                    .unwrap();
+                println!(
+                    "{}",
+                    serde_json::to_string_pretty(&canonical.0.cids()).unwrap()
+                );
+            }
+            Self::Message { cid } => {
+                let cid: Cid = cid.parse().unwrap();
+                let client = new_client();
+
+                let msg = messages(client, cid)
+                    .await
+                    .map_err(|e| {
+                        stringify_rpc_err(e);
+                    })
+                    .unwrap();
+                println!("{}", serde_json::to_string_pretty(&msg).unwrap());
+            }
+            Self::ReadObj { cid } => {
+                let cid: Cid = cid.parse().unwrap();
+                let client = new_client();
+
+                let obj = read_obj(client, cid)
+                    .await
+                    .map_err(|e| {
+                        stringify_rpc_err(e);
+                    })
+                    .unwrap();
+                println!("{}", serde_json::to_string_pretty(&obj).unwrap());
+            }
+        }
+    }
+}
+
+fn stringify_rpc_err(e: JsonRpcError) {
+    match e {
+        JsonRpcError::Full {
+            code,
+            message,
+            data: _,
+        } => {
+            return warn!("JSON RPC Error: Code: {} Message: {}", code, message);
+        }
+        JsonRpcError::Provided { code, message } => {
+            return warn!("JSON RPC Error: Code: {} Message: {}", code, message);
+        }
+    }
+}
```

Comment:
Feel like this should a general function, maybe move to  forest/src/cli/mod.rs

Reply:
this function also doesn't do what it says it does, should either return a string and swap the `warn!` for a `format!` or change this name